# --- T2-COPYRIGHT-NOTE-BEGIN ---
# This copyright note is auto-generated by ./scripts/Create-CopyPatch.
# 
# T2 SDE: package/.../vdr-softdevice/hotfix-offset_t.patch
# Copyright (C) 2008 The T2 SDE Project
# 
# More information can be found in the files COPYING and README.
# 
# This patch file is dual-licensed. It is available under the license the
# patched project is licensed under, as long as it is an OpenSource license
# as defined at http://www.opensource.org/ (e.g. BSD, X11) or under the terms
# of the GNU General Public License as published by the Free Software
# Foundation; either version 2 of the License, or (at your option) any later
# version.
# --- T2-COPYRIGHT-NOTE-END ---

Index: softdevice-cvs-20080922/mpeg2decoder.c
===================================================================
--- softdevice-cvs-20080922.orig/mpeg2decoder.c
+++ softdevice-cvs-20080922/mpeg2decoder.c
@@ -1016,7 +1016,7 @@ static int read_packet_RingBuffer(void *
 };
 
 #if LIBAVFORMAT_BUILD >4625
-static offset_t seek_RingBuffer(void *opaque, offset_t offset, int whence)
+static int64_t seek_RingBuffer(void *opaque, int64_t offset, int whence)
 #else
 static int seek_RingBuffer(void *opaque, offset_t offset, int whence)
 #endif
@@ -1135,7 +1135,7 @@ start:
     return 0;
 };
 
-int cMpeg2Decoder::seek(offset_t offset, int whence) {
+int cMpeg2Decoder::seek(int64_t offset, int whence) {
    printf("unimplemented: seek offset %lld whence %d\n", (long long int)offset, whence);
    return -EINVAL;
 };
Index: softdevice-cvs-20080922/mpeg2decoder.h
===================================================================
--- softdevice-cvs-20080922.orig/mpeg2decoder.h
+++ softdevice-cvs-20080922/mpeg2decoder.h
@@ -312,7 +312,7 @@ private:
     bool          packetMode;
 public:
     int read_packet(uint8_t *buf, int buf_size);
-    int seek(offset_t offset, int whence);
+    int seek(int64_t offset, int whence);
 
 public:
     cMpeg2Decoder(cAudioOut *AudioOut, cVideoOut *VideoOut);
